turbine/src/broadcast_stage/broadcast_utils.rs

```diff
@@ -44,16 +44,25 @@ fn keep_coalescing_entries(
     max_tick_height: u64,
     serialized_batch_byte_count: u64,
     max_batch_byte_count: u64,
+    process_stats: &mut ProcessShredsStats,
 ) -> bool {
-    // This slot is not over.
-    last_tick_height < max_tick_height &&
-    // We have not exceeded max batch byte count.
-    serialized_batch_byte_count < max_batch_byte_count &&
-    {
-        let bytes_to_fill_erasure_batch = data_shred_bytes_per_batch() - (serialized_batch_byte_count % data_shred_bytes_per_batch());
-        // We haven't tightly packed this batch.
-        (bytes_to_fill_erasure_batch as f64) > target_batch_pad_bytes()
+    if last_tick_height >= max_tick_height {
+        // The slot has ended.
+        process_stats.coalesce_exited_slot_ended += 1;
+        return false;
+    } else if serialized_batch_byte_count >= max_batch_byte_count {
+        // Exceeded the max batch byte count.
+        process_stats.coalesce_exited_hit_max += 1;
+        return false;
+    }
+    let bytes_to_fill_erasure_batch =
+        data_shred_bytes_per_batch() - (serialized_batch_byte_count % data_shred_bytes_per_batch());
+    if (bytes_to_fill_erasure_batch as f64) < target_batch_pad_bytes() {
+        // We're close enough to tightly packing erasure batches. Just send it.
+        process_stats.coalesce_exited_tightly_packed += 1;
+        return false;
     }
```

Comment:
may not be a huge deal but we could avoid using floating point calculations/conversions if we just do (or equivalent w/ helper functions):
```suggestion
    if bytes_to_fill_erasure_batch * 20 < data_shred_bytes_per_batch() {
        // We're close enough to tightly packing erasure batches. Just send it.
        process_stats.coalesce_exited_tightly_packed += 1;
        return false;
    }
```
^ may be worth it since we're looking at leader side of things

Reply:
Let me know if you guys prefer the latest method that uses OnceLock for handling the `TARGET_BATCH_BYTES_DEFAULT` and `TARGET_BATCH_PAD_BYTES`

Reply:
EDIT: Brennan, seeing your new update now. ignore above...